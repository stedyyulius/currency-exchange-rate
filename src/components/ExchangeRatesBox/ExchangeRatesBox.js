import React from 'react';
import { connect } from 'react-redux';

import styles from './ExchangeRatesBox.module.css'

const ExchangeRatesBox = props => {
    return (
        <div className={styles.container}>
            <div className={styles.rateDetails}>
                <div className={styles.rate}>
                    <p>{props.rate.name}</p>
                    <p>{props.rate.value * props.rateValue}</p>
                </div>
                <div className={styles.rateName}>
                    <i>{props.rate.name}</i>
                </div>
                <i>1 {props.base} - {props.rate.name} {props.rate.value}</i>
            </div>
            <div className={styles.deleteButton}>
                <p onClick={() => props.removeCurrency(props.rate.name)}>(-)</p>
            </div>
        </div>
    )
}

const mapStateToProps = state => {
    return {

    }
}

const mapDispatchToProps = dispatch => {
    return {

    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExchangeRatesBox);