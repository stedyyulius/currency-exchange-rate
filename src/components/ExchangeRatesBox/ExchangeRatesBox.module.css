.container {
  border: var(--themeBorder);
  display: inline-block;
  display: flex;
}

.rateDetails {
  text-align: left;
  width: 92%;
  padding: 5px;
}

.rate {
  display: flex;
  justify-content: space-between;
  line-height: 0%;
}

.rateName {
  font-weight: bold;
}

.deleteButton {
  border-left: var(--themeBorder);
  height: auto;
  width: 8%;
  padding-top: 14px;
  font-size: 18px;
}
