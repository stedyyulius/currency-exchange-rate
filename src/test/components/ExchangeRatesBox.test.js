import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { shallow } from 'enzyme';

import ExchangeRatesBox from '../../components/ExchangeRatesBox/ExchangeRatesBox';
import store from '../../store';

configure({ adapter: new Adapter() });

describe('ExchangeRatesBox', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<ExchangeRatesBox store={store} />).dive();
        expect(wrapper).toMatchSnapshot();
    });
});