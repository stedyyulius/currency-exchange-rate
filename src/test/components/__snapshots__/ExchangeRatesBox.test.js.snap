// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRatesBox should render correctly 1`] = `
<ExchangeRatesBox
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
/>
`;
